github/client.go

```diff
@@ -37,6 +44,14 @@ type Client struct {
 	cachedClient *simpleClient
 }
 
+func SetUserAgent() {
+	if userAgent := os.Getenv("HUB_USERAGENT"); userAgent != "" {
+		UserAgent = userAgent
+	} else if userAgent, err := git.Config("hub.useragent"); err == nil {
+		UserAgent = userAgent
```

Comment:
Does your Enterprise instance also require setting a custom user agent string?

Reply:
This makes sence, I thought that is the default pattern.